staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -184,12 +184,15 @@ func (s *SpdyRoundTripper) dialWithHttpProxy(req *http.Request, proxyURL *url.UR
 
 	//nolint:staticcheck // SA1019 ignore deprecated httputil.NewProxyClientConn
 	proxyClientConn := httputil.NewProxyClientConn(proxyDialConn, nil)
-	_, err = proxyClientConn.Do(&proxyReq)
+	response, err := proxyClientConn.Do(&proxyReq)
 	//nolint:staticcheck // SA1019 ignore deprecated httputil.ErrPersistEOF: it might be
 	// returned from the invocation of proxyClientConn.Do
 	if err != nil && err != httputil.ErrPersistEOF {
 		return nil, err
 	}
+	if response != nil && response.StatusCode != http.StatusOK {
```

Comment:
limiting this to exactly 200 does not seem correct... change this to accept any 2xx status code, per https://www.rfc-editor.org/rfc/rfc7231#section-4.3.6:

>   Any 2xx (Successful) response indicates that the sender (and all
>   inbound proxies) will switch to tunnel mode immediately after the
>   blank line that concludes the successful response's header section;
>   data received after that blank line is from the server identified by
>   the request-target.  Any response other than a successful response
>   indicates that the tunnel has not yet been formed and that the
>   connection remains governed by HTTP.

Reply:
you're right, thanks for catching that